code/modules/power/heavy_emitter.dm

```diff
@@ -0,0 +1,261 @@
+				. =  FALSE
+			linked_cannon = object
+
+/obj/machinery/power/heavy_emitter/centre/should_have_node()
+	return anchored
+
+/obj/machinery/power/heavy_emitter/centre/connect_to_network()
+	if(!anchored)
+		return FALSE
+	. = ..()
+
+/obj/machinery/power/heavy_emitter/centre/turn_on()
+	if(!is_fully_constructed)
+		return linked_interface.turn_off()
+	use_power = IDLE_POWER_USE
+	firing = TRUE
+	icon_state = "centre"
+
+/obj/machinery/power/heavy_emitter/centre/turn_off()
+	firing = FALSE
+	icon_state = "centre_off"
+
+/obj/machinery/power/heavy_emitter/centre/process(delta_time)
+	if(!firing || machine_stat & BROKEN || surplus() < active_power_usage)
+		return
+	if(!check_part_connectivity())
+		return linked_interface.turn_off()
+
+	timer += delta_time
+
+	add_load(idle_power_usage)
+	if(timer >= max_timer)
+		timer = 0
+		add_load(active_power_usage)
+		radiation_pulse(src,500,can_contaminate=FALSE)
+		visible_message("<span class='notice'>Heavy Emitter Core is powering the cannon....</span>")
+		INVOKE_ASYNC(linked_cannon,/obj/machinery/power/heavy_emitter/cannon.proc/fire)
+		heat += 500
+
+	if(heat > max_heat)
+		explosion(src,4,8,16)
+		return
+
+	for(var/V in vents)
+		if(heat <= 0 || !V)
+			break
+		var/obj/machinery/power/heavy_emitter/vent/vent = V
+		if(vent.vent_gas())
+			heat -= 100
+
+
+/obj/machinery/power/heavy_emitter/arm
+	name = "Seismic Stabilizer Arm"
+	desc = "Dampens the recoil from firing to virtually nothing"
+	icon_state = "arm"
+
+/obj/machinery/power/heavy_emitter/interface
+	name = "Kinetic Amplification Manipulation Interface"
+	desc = "Allows for control over the core."
+	icon_state = "interface_off"
+	///Core connected to this thing
+	var/obj/machinery/power/heavy_emitter/centre/connected_core
+
+/obj/machinery/power/heavy_emitter/interface/attack_hand(mob/living/user)
+	. = ..()
+	var/turf/T = get_step(src,turn(dir,180))
+	var/obj/machinery/power/heavy_emitter/centre/centre = locate() in T
+
+	if(!centre || !centre.check_part_connectivity())
+		turn_off()
+		return
+
+	connected_core = centre
+
+	if(connected_core.firing)
+		to_chat(user, "<span class='warning'>You power on the Heavy Emitter!</span>")
+		turn_off()
+	else
+		to_chat(user, "<span class='warning'>You disable the Heavy Emitter!</span>")
+		turn_on()
+
+
+/obj/machinery/power/heavy_emitter/interface/turn_on()
+	icon_state = "interface"
+	connected_core.turn_on()
+
+/obj/machinery/power/heavy_emitter/interface/turn_off()
+	icon_state = "interface_off"
+	connected_core.turn_off()
+
+/obj/machinery/power/heavy_emitter/vent
+	name = "Energy Core Vent"
+	desc = "Circulates air around the core, preventing it from overheating."
+	icon_state = "vent"
+
+/obj/machinery/power/heavy_emitter/vent/proc/vent_gas()
+	var/turf/open/open_turf = get_step(src,dir)
+	if(!istype(open_turf))
+		return FALSE
+	var/datum/gas_mixture/gases = open_turf.return_air()
+	//Space can exist
+	if(gases)
+		gases.temperature += 100
```

Comment:
putting this in space would negate it's downsides right? Also, return_air() should return a valid mix from space tiles, so your check will do little here.

Another thing, this would heat up all mixes equally, which is kinda cringe ngl, you'll want to use something that takes heat capacity into account, look into temperature_share(), it's built to allow for stuff like this. you can just, not pass in a sharer gasmix and it'll adapt/overcome. Consider storing the core temperature in kelvin, and as it actually would be, and having it actually share with the air around it, so you could blow up the emitter by lighting it on fire.